util/type_traits.h

```diff
@@ -254,8 +254,8 @@ struct addition : std::false_type {};
 
 template <typename LeftOperand, typename RightOperand>
 struct addition<LeftOperand, RightOperand,
-                std::void_t<decltype(std::declval<LeftOperand>() +
-                                     std::declval<RightOperand>())>>
+                std::void_t<decltype(std::declval<LeftOperand&>() +
+                                     std::declval<RightOperand&>())>>
```

Comment:
This is a strange change, why is it needed? I'm guessing because the type is not defined yet? If that's the case, this is probably preferable:
```suggestion
                std::void_t<decltype(std::declval<const LeftOperand&>() +
                                     std::declval<const RightOperand&>())>>
```

Reply:
Actually only change for LeftOperand is needed, removed others.
It is needed similarly to reference on line 242 - to handle rval of first operand for subtraction operator.

Reply:
The problem I see is that binary addition is a non-modying operation, so a non-const reference should not be needed or requested.

Reply:
@ProGTX You are right, it shouldn't work like that. Removed this change and revert enabling this test for now.